src/Groups/GAPGroups.jl

```diff
@@ -1701,6 +1701,40 @@ function length(g::FPGroupElem)
 end
 
 
+"""
+    (G::FPGroup)(l::AbstractVector{Pair{T, S}}) where {T <: IntegerUnion, S <: IntegerUnion}
+
+Return the element `x` of `G` that is described by `l`
+in the sense that `x` is the product of the powers `gen(G, i_j) ^ e_j`
+where the `l[j]` is equal to `i_j => e_j`.
+If the `i_j` in adjacent entries of `l` are different and the `e_j` are
+nonzero then `l` is the vector of syllables of `x`, see [`syllables`](@ref).
+
+# Examples
+```jldoctest
+julia> G = free_group(2);  l = [1 => 3, 2 => -1];
+
+julia> x = G(l)
+f1^3*f2^-1
+
+julia> syllables(x) == l
+true
+```
+"""
+function (G::FPGroup)(l::AbstractVector{Pair{T, S}}) where {T <: IntegerUnion, S <: IntegerUnion}
+   n = ngens(G)
+   ll = S[]
+   for p in l
+     0 < p.first && p.first <= n || throw(ArgumentError("generator number is at most $n"))
+     push!(ll, p.first)
+     push!(ll, p.second)
```

Comment:
`ll` is a `Vector{S}` but `first` is of type `T`. I guess in practice `T` and `S` both will be `Int` and this won't matter... But we could also sidestep all of this by changing `ll` into a `Vector{GAP.Obj}` and then doing this:
```suggestion
     push!(ll, GAP.Obj(p.first))
     push!(ll, GAP.Obj(p.second))
```
And then of course later we can just write `GapObj(ll)` w/o a need for recursive conversion.